apps/dojo/src/contexts/url-params-context.tsx

```diff
@@ -34,22 +34,27 @@ interface URLParamsProviderProps {
   children: ReactNode;
 }
 
-export function URLParamsProvider({ children }: URLParamsProviderProps) {
-  const router = useRouter();
-  const pathname = usePathname();
-  const searchParams = useSearchParams();
-
-  // Initialize state from URL params
-  const [state, setState] = useState<URLParamsState>(() => ({
+function generateURLParamsState(searchParams: URLSearchParams): URLParamsState {
+  return {
     view: (searchParams.get("view") as View) || "preview",
     sidebarHidden: searchParams.get("sidebar") === "false",
     chatDefaultOpen: searchParams.get("chatDefaultOpen") !== "false",
     frameworkPickerHidden: searchParams.get("frameworkPicker") === "false",
     viewPickerHidden: searchParams.get("viewPicker") === "false",
     featurePickerHidden: searchParams.get("featurePicker") === "false",
+    file: searchParams.get("file") || undefined,
```

Comment:
@tylerslaton this line you added/edited two months ago was only in one of the two blocks. Just want to make sure that you think it's fine that I include it here where it will now be included in both places that invoke `generateURLParamsState`.